tests/utils/test.py

```diff
@@ -160,3 +168,126 @@ def create_fake_postcode():
     # Corsica and some DROM-COM department being 3 digits long,
     # return value may be longer than 5. Cut it.
     return f"{department}{int(random.randint(0, 999)):03}"[:5]
+
+
+class _AssertSnapshotQueriesContext(CaptureQueriesContext):
+    def __init__(self, snapshot, connection):
+        self.tx_id = 0
+        self.snapshot = snapshot
+        super().__init__(connection)
+
+    def normalize_sql(self, sql):
+        if re.match(r'^SAVEPOINT +".+?" *$', sql):
+            self.tx_id += 1
+            # 'SAVEPOINT "s124109847980928_x70"'
+            return f'SAVEPOINT "<{self.tx_id}>"'
+        if re.match(r'^RELEASE +SAVEPOINT +".+?" *$', sql):
+            # 'RELEASE SAVEPOINT "s124109847980928_x69"'
+            content = f'RELEASE SAVEPOINT "<{self.tx_id}>"'
+            self.tx_id -= 1
```

Comment:
Comme:
```
postgres=!# COMMIT;
ROLLBACK
postgres=# BEGIN;
BEGIN
postgres=*# SAVEPOINT "a";
SAVEPOINT
postgres=*# SAVEPOINT "b";
SAVEPOINT
postgres=*# RELEASE SAVEPOINT "a";
RELEASE
```
marche, ta version simplifiée peut potentiellement induire en erreur donc je préfère éviter en l'état...

Il faudrait maintenir la correspondance `{"s124109847980928_x70": "<1>", "s124109847980928_x69": "<2>"}` pour faire bien les choses (ce qui ne nécessite pas tant de changement que cela :sweat_smile: )

Reply:
Oui, mais en fait non, parce que :
```sql
itou3=# BEGIN;
BEGIN
itou3=*# SAVEPOINT "a";
SAVEPOINT
itou3=*# SAVEPOINT "a";
SAVEPOINT
itou3=*# RELEASE SAVEPOINT "a";
RELEASE
itou3=*# RELEASE SAVEPOINT "a";
RELEASE
```

Donc, il faudrait un truc du genre:
```python
{
    "s124109847980928_x70": ["<1>"],
}
```
Avec `append` et `pop` lors des release. Et puis éventuellement des rollback to qui revert un savepoint intermédiaire.

Autant ne pas gérer les savepoints :sweat: